### PR TITLE
Feature/wifi 1350

### PR DESCRIPTION
**Pull request by 4c74356b41**
matrix builds

### CODE REVIEW

**Comment by 4c74356b41** [comment]
thats something they should provide, i cant fix their systems for them

File: .github/workflows/nightly.yml [hunk 1]
```diff
@@ -136,12 +205,59 @@ jobs:
...
-        echo "Grafana Link: https://grafana.lab.wlan.tip.build/TO_BE_PROVIDED"
-        echo "Kibana Link: https://kibana.lab.wlan.tip.build/TO_BE_PROVIDED"
-      shell: bash
-    - name: Publish Unit Test Results
+    - name: Upload Unit Test Results
+      if: always()
+      uses: actions/upload-artifact@v2
+      with:
+        name: Unit Test Results ( nola-${{ matrix.number }} )
+        path: /ci/test_everything.xml
```
Comment (leonidlm) [inline .github/workflows/nightly.yml:213]: Maybe we should name the file with a `matrix.number` postfix and the date? Not sure where this is uploaded to, however.

File: .github/workflows/nightly.yml [hunk 2]
```diff
@@ -136,12 +205,59 @@ jobs:
...
+      matrix: ${{ fromJson( needs.generate-matrix.outputs.matrix ) }}
+    if: always()
+
+    steps:
+    - name: Post links to PR
+      uses: unsplash/comment-on-pr@master
+      env:
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      with:
+        msg: "Grafana Link: https://grafana.lab.wlan.tip.build/TO_BE_PROVIDED\n Kibana Link: https://kibana.lab.wlan.tip.build/TO_BE_PROVIDED"
```
Comment (leonidlm) [inline .github/workflows/nightly.yml:230]: :heart: NICE!

**Review by leonidlm — APPROVED** [review]
Looks good to me!

File: .github/workflows/nightly.yml (hunk 1 above)
Comment (4c74356b41) [reply .github/workflows/nightly.yml]: these are only uploaded temporary, to be consumed by the real upload results task, but I'm not sure and we can probably do what you advice anyway. shall be better if we can actually retrieve them

File: .github/workflows/nightly.yml [hunk 3]
```diff
@@ -136,12 +205,59 @@ jobs:
...
+
+    steps:
+    - name: Fetch kubeconfig
+      shell: bash
+      run: |
+        aws eks update-kubeconfig  --name ${{ env.AWS_EKS_NAME }}
+    - name: Remove temporary Cloud SDK instances
+      shell: bash
+      run: |
+        helm uninstall tip --namespace nola-${{ matrix.number }}
```
Comment (brennerm) [inline .github/workflows/nightly.yml:262]: Please also delete the namespace.

File: .github/workflows/nightly.yml [hunk 4]
```diff
@@ -1,43 +1,75 @@
...
+      "number": "04",
+      "ip_address": "10.28.3.12",
+      "ap_model": "ECW5410",
+      "cleanup": true
+    }
+  ]'
 
 on:
   workflow_dispatch:
+  pull_request:
```
Comment (brennerm) [inline .github/workflows/nightly.yml:41]: Is this just for testing or will it get merged?

File: .github/workflows/nightly.yml [hunk 5]
```diff
@@ -67,6 +103,7 @@ jobs:
         path: toolsmith
         repository: Telecominfraproject/Toolsmith
         token: ${{ secrets.PAT_TOKEN }}
+        ref: feature/WIFI-1350
```
Comment (brennerm) [inline .github/workflows/nightly.yml:106]: Don't forget to remove this reference before merging.

File: .github/workflows/nightly.yml (hunk 4 above)
Comment (4c74356b41) [reply .github/workflows/nightly.yml:41]: i think it shall get merged, why?

File: .github/workflows/nightly.yml (hunk 1 above)
Comment (4c74356b41) [reply .github/workflows/nightly.yml]: these are only available under the tests they were uploaded at, so tagging them with day probably doesn't make a lot sense? 
image

File: .github/workflows/nightly.yml (hunk 4 above)
Comment (4c74356b41) [reply .github/workflows/nightly.yml:41]: i'm not sure we want to run a different set of test nightly\on pr

Comment (brennerm) [reply .github/workflows/nightly.yml:41]: Was just wondering cause we didn't trigger the nightly tests for PRs beforehand and I don't see a reason why this changed. But I'm completely fine with doing so.

Comment (jaspreetsachdev) [reply .github/workflows/nightly.yml:41]: With pytest we should be able to call different set of test cases if we want to that may differ from a nightly set. The idea is PR tests are a quick sanity and nightly for longer sanity.

**Comment by 4c74356b41** [comment]
this is superseeded by the existing build: https://github.com/Telecominfraproject/wlan-testing/actions/workflows/cloud-controller-build.yaml
leaving branch for reference and closing this PR

**Pull request closed without merging**